pkg/shares/share_splitting.go

```diff
@@ -45,6 +45,11 @@ func Split(data coretypes.Data, useShareIndexes bool) ([]Share, error) {
 			BlobSharesUsed(len(data.Blobs[0].Data)),
 			int(data.SquareSize),
 		)
+		// force msgIndexes to be the first share index
+		if len(blobIndexes) != 0 && useShareIndexes {
+			blobShareStart = int(blobIndexes[0])
+		}
```

Comment:
Are we able to specifically add a unit test for this?

Reply:
:+1: yeah defo needed, thanks. fails on main
https://github.com/celestiaorg/celestia-app/pull/1111/commits/c67976551ac4cebc636bef35ea73ff1ffca65bb2

also https://github.com/celestiaorg/celestia-app/issues/1110#issuecomment-1352389547